### DISCUSSION TITLE
Filter response only for authenticated user

### DISCUSSION THREAD

**Discussion by adamszeptycki** [prisma/prisma#9210] [Q&A]
Hey, 
Is it possible to filter response to include only items currently authenticated user has access to?

I've seen this: https://www.prisma.io/blog/graphql-directive-permissions-authorization-made-easy-54c076b5368e
But it's not what I want.

Let's say I have this example:
```prisma
model User {
  id       String  @id @default(cuid())
  email    String  @unique
  password String 
  name     String? 
  posts    Post[]
}
// Define the `Post` table in the database
model Post {
  id       String  @id @default(cuid())
  title    String
  content  String?
  authorId String
  author   User
}
```

If user executes a request:

```graphql
query {
listPost {
title
}
}
```
I would like to return ONLY posts created by this user. What is the best way to achieve that?

**Comment by adamszeptycki** [comment]
I'm using ` prisma-appsync`

**Reply by ryands17** [reply]
You can do this by fetching the user's `id` from the `context` that AppSync provides and then add a condition on the query that fetches only the posts that are of that specific user.

**Comment by blevine** [comment]
Are you asking whether Prisma can do this for you without you having to implement the SELECT yourself?

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
